src/movepick.cpp

```diff
@@ -63,9 +63,8 @@ MovePicker::MovePicker(const Position& p, Move ttm, Depth d, const ButterflyHist
 
   assert(d > 0);
 
-  stage = pos.checkers() ? EVASION_TT : MAIN_TT;
-  ttMove = ttm && pos.pseudo_legal(ttm) ? ttm : MOVE_NONE;
-  stage += (ttMove == MOVE_NONE);
+  stage = pos.checkers() ? EVASION_INIT : CAPTURE_INIT;
+  useTTM = ((ttMove = ttm && pos.pseudo_legal(ttm) ? ttm : MOVE_NONE));
```

Comment:
are all parenthesis needed here?

Reply:
If I remove the extra set, the compiler complains about using an assignment in a truth check or something like that.

Reply:
I was able to remove them.